state/dashboard/operations/__tests__/trendingMovies.test.js

```diff
@@ -0,0 +1,79 @@
+import trendingMovies from '../trendingMovies';
+import * as types from '../../types';
+import { trendingMoviesResponseSuccess, trendingMoviesResponseError } from '../__mocks__/trendingMovies';
+import normalizedMovies from '../../normalize/movies';
+import { fetchMoviesSuccess, fetchMoviesError } from '../../actions';
+import { mergeMoviesList } from '../../../data/actions';
+
+jest.mock('../../../data/actions', () => ({
+  mergeMoviesList: jest.fn()
+}));
+jest.mock('../../actions', () => ({
+  fetchMoviesSuccess: jest.fn(),
+  fetchMoviesError: jest.fn()
+}));
+
+describe('trendingMovies logic', () => {
+  const action = {
+    type: types.FETCH_TRENDING_MOVIES,
+    page: 1
+  };
+  const mockHttpClient = ({ method, response, reject } = { reject: false, response: {} }) => {
+    const httpClient = {
+      [method]: () => new Promise((resolve, deny) => {
+        if (reject) {
+          deny(response);
+        } else {
+          resolve(response);
+        }
+      })
+    };
+    jest.spyOn(httpClient, method);
+    return httpClient;
+  };
+  const dispatch = jest.fn();
+  const done = jest.fn();
+
+  it('matches snapshot', () => {
+    expect(trendingMovies).toMatchSnapshot();
+  });
+
+  describe('success', () => {
+    const { page } = action;
+    const currentPage = page;
+    const httpClient = mockHttpClient({ method: 'get', response: trendingMoviesResponseSuccess });
+    trendingMovies.process({ httpClient, action }, dispatch, done);
+
+    it('httpClient called with right arguments', () => {
```

Comment:
I think this test should be outside of success block, because it is related to both scenarios - success and failure.

Reply:
I am not sure. If we have right arguments than we have never to get on failure section.
Can disscuss.